### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by MihailNenahov**
Writing integration tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,6 +20,34 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
```
Comment (arramanenka) [inline pom.xml]: Junit is included in spring boot test dependency

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Testcontainers
+class LogisticControllerTest {
+    @ClassRule
+    private DockerComposeContainer compose = new DockerComposeContainer(
+            new File("test/resources/dockerCompose.yml"));
```
Comment (arramanenka) [inline src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: Please try to rewrite it without using test/resources

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Test
+    void findOrders() {
+        String request = "http://" + compose.getServiceHost("service_l", 8080) + ":" +
```
Comment (arramanenka) [inline src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: Use mockmvc/webtestclient

**Review by arramanenka — CHANGES_REQUESTED** [review]

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java (hunk 3 above)
Comment (MihailNenahov) [reply src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: done

File: pom.xml (hunk 1 above)
Comment (MihailNenahov) [reply pom.xml]: removed

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java (hunk 2 above)
Comment (MihailNenahov) [reply src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: still problem with configuring classpath

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @Test
+    void findOrders() {
+       this.webTestClient.
+               get().
+               uri("/orders?customerName=Misha&page=0&size=1").
```
Comment (arramanenka) [inline src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: I would avoid hardcoding values like that. Webtestclient supports map for parameters

Comment (arramanenka) [reply src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: Also you have almost identical tests for searching with customerName and without it. Can you verify in response that this customerName indeed matches one in response?

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                isNotEmpty().
+                jsonPath("[0].merchandiseQuantity").
+                isNotEmpty().
+                jsonPath("[0].position").
+                isNotEmpty();
+    }
+
+    @Test
+    void saveNewCustomer() {
+        this.webTestClient.
```
Comment (arramanenka) [inline src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: Also, for all of these methods you have a very weird formatting. Dot should be before the method call, not after it

File: src/test/java/com/logistic/project/util/CoordinateCalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.org.apache.commons.lang.ObjectUtils;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CoordinateCalculatorTest {
+
+    @Test
+    void getCoordinateDistance() {
```
Comment (arramanenka) [inline src/test/java/com/logistic/project/util/CoordinateCalculatorTest.java]: This is not enough of a test. It only covers one positive scenario. I could have as well hardcoded 5 into return value of getCoordinateDistance and your tests would pass

File: src/test/java/com/logistic/project/controller/LogisticControllerTest.java [hunk 7]
```diff
@@ -47,97 +47,101 @@ static void setProperties(DynamicPropertyRegistry registry) {
...
-                expectBody().
-                jsonPath("name").
-                isNotEmpty().
-                jsonPath("position").
-                isNotEmpty();
+        this.webTestClient
+                .post()
+                .uri("/customer")
+                .header("Content-Type", "application/json")
+                .body(BodyInserters.fromValue(Customer.builder().name("smth").position(new Coordinate(1, 2)).build()))
```
Comment (arramanenka) [inline src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: try without bodyinserters (maybe method like bodyValue)

Comment (MihailNenahov) [reply src/test/java/com/logistic/project/controller/LogisticControllerTest.java]: done

**Review by arramanenka — APPROVED** [review]

**Pull request merged**